crates/bevy_ecs/src/system/query.rs

```diff
@@ -292,6 +292,31 @@ where
         }
     }
 
+    /// Gets the result of a single-result query
+    ///
+    /// If the query has exactly one result, returns the result inside `Ok`
+    /// otherwise returns either `Err(QuerySingleError::NoEntities(...))`
+    /// or `Err(QuerySingleError::MultipleEntities(...))`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///  # use bevy_ecs::system::{Query, QuerySingleError};
+    ///  # use bevy_render::camera::Camera;
+    /// fn do_something_with_camera(query: Query<&Camera>) {
```

Comment:
Can we add a check that this function is a system, by using `.system()` on it in a hidden position

Actually, do we have any tests for this api in general - maybe adding a full test would be useful, if you don't mind.

Reply:
I see no reason to delay merging this now. Feel free to follow up with full tests if you feel inclined.